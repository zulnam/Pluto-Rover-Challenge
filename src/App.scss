#root {
  display: grid;
  grid-template-columns: 1fr;
  margin: 20px;

  @media (min-width: 560px) {
    grid-template-columns: 1fr 2fr;
}
}

.commandContainer {
  h3 {
    text-align: center;
  }

  form {
    display: flex;
    justify-content: space-around;

    button {
      border-radius: 8px;
      width: 64px;
      height: 32px;
    }
  }
}

.roverPositionContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;

  p {
    text-align: center;
    margin: 0;
  }

  :nth-child(4), :nth-child(5), :nth-child(6) {
    border: 1px solid black;
    padding: 12px;
  }
} 

.commandHistoryContainer {
  ul {
    list-style-type: none;

    :first-child {
      background-color: aquamarine;
      opacity: 1;
    }
    
    li {
      opacity: 0.5;
    }
  }
}
