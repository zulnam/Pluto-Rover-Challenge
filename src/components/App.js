import React, { useState, useReducer, useEffect } from 'react';
import movement from '../reducers/movement';
import MovementContext from '../context/movementContext';
import Title from './Title/Title';
import CommandCenter from './CommandCenter/CommandCenter';
import ActivityLog from './ActivityLog/ActivityLog';

const initialCoordinates = {
  X: 0,
  Y: 0,
  direction: 'N',
};

const App = () => {
  const [coordinates, setCoordinates] = useReducer(
    movement,
    initialCoordinates
  );
  const [items, setItems] = useState([]);
  const [obstacleCoordinates, setObstacleCoordinates] = useState([]);

  useEffect(() => {
    const preliminaryScan = Math.floor(Math.random() * 100);

    for (let i = 1; i <= preliminaryScan; i++) {
      const newObstacle = {
        X: Math.floor(Math.random() * 100),
        Y: Math.floor(Math.random() * 100),
      };

      setObstacleCoordinates((obstacleCoordinates) => [
        ...obstacleCoordinates,
        newObstacle,
      ]);
    }
    /** excluded eslint rule for obstacleCoordinates dependency as
     * this is the boot spawn generator for obstacles
     */
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <MovementContext.Provider
      value={{
        coordinates,
        setCoordinates,
        items,
        setItems,
        obstacleCoordinates,
      }}
    >
      <Title />
      <CommandCenter />
      <ActivityLog />
    </MovementContext.Provider>
  );
};

export default App;
