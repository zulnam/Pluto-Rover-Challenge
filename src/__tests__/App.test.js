import { render } from '@testing-library/react';
import App from '../components/App';

it('renders the main app', () => {
  const div = document.createElement('div');
  const { container } = render(<App />, div);
  expect(container.firstChild).toMatchSnapshot();
});
