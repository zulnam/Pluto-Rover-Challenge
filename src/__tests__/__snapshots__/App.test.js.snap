// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the main app 1`] = `
<h1
  class="css-131o7kf"
>
  Rover Command
</h1>
`;
